src/strategyHelpers/canary/podCanaryHelper.ts

```diff
@@ -21,49 +21,57 @@ export async function deployPodCanary(
       throw Error('Percentage must be between 0 and 100')
 
    for (const filePath of filePaths) {
-      const fileContents = fs.readFileSync(filePath).toString()
-      const parsedYaml = yaml.safeLoadAll(fileContents)
+      const fileContents = fs.readFileSync(filePath, 'utf8')
+      const parsedYaml = yaml.loadAll(fileContents)
       for (const inputObject of parsedYaml) {
-         const name = inputObject.metadata.name
-         const kind = inputObject.kind
+         if (
+            inputObject &&
+            typeof inputObject === 'object' &&
+            'metadata' in inputObject &&
+            'kind' in inputObject
```

Comment:
💡 thoughts, just add for clarity comments: type assertions to avoid casting to any. here you re checking if string "kind" exist of is it more to this please?

Reply:
agreed instead of `any` we can use `{metadata: {name?: string}, kind: string}` and then add `typeof` guards for the types of metadata, metadata.name, and kind?